### PR TITLE
When a record changes, make sure any new URIs do not collide with exi…

### PR DESCRIPTION
**Pull request by jannistsiroyannis**
…sting URIs

Now that it is possible to change a records URI, we must make sure that
the changed URI does not conflict with an existing one.

### CODE REVIEW

File: whelk-core/src/main/groovy/whelk/Whelk.groovy [hunk 1]
```diff
@@ -352,6 +353,13 @@ class Whelk {
             preUpdateDoc = doc.clone()
             updateAgent.update(doc)
             normalize(doc)
+
+            boolean detectCollisionsOnTypedIDs = false
+            List<Tuple2<String, String>> collidingIDs = getIdCollisions(doc, detectCollisionsOnTypedIDs)
+            if (!collidingIDs.isEmpty()) {
+                log.info("Refused update of " + doc.getShortId() + ". Document would have ID collision with : " + collidingIDs)
+                throw new WhelkRuntimeException("Update of ${doc.getShortId()} failed, due to ID collision with : " + collidingIDs)
+            }
```
Comment (olovy) [inline whelk-core/src/main/groovy/whelk/Whelk.groovy:362]: Maybe extract this to a method `checkIdCollsions(doc)` and use it in all three places?

Comment (jannistsiroyannis) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy:362]: But the error messages are slighlty different, depending on if its CREATE or UPDATE.

Comment (olovy) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy:362]: Do they have to be?
"`StorageCreateFailedException`" is not referenced in any other place.

Comment (jannistsiroyannis) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy:362]: It's referenced here: rest/src/main/groovy/whelk/rest/api/Crud.groovy:729, apparently so we can return a http 409: Conflict.
I don't mean to say that's necessarily a meaningful thing to do, but I'd rather not change the crud interface if I don't have to.

Comment (jannistsiroyannis) [reply whelk-core/src/main/groovy/whelk/Whelk.groovy:362]: HAha, det behövs inte! PostgreSQL.verifyDocumentIdRetention() sköter redan det här. Förmodligen upkom den här i ett tidigare skede (den första stora inläsningen)!

**Pull request closed without merging**